json: update validator interface

IoannisRP opened the pull request:
rapidjson works on its own internal string representation.
A string view is more appropriate than a string here.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none

IoannisRP committed:
json: update validator interface

rapidjson works on its own internal string representation.
A string view is more appropriate than a string here.

BenPope reviewed (commented):
Is it worth changing the license schemas (and any other users of this interface) to be `constexpr string_views` now, so that `strlen` isn't needed?

IoannisRP committed:
json: update validator users

Update all clients of the json validator to not create a string
but a string_view of the string literal.

IoannisRP commented:
All unit tests were executed locally with sanitizers turned on as an extra precaution.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--dd0648de-db30-4999-ae8c-a23fe45c4ae3-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/69280">build#69280</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "chunked_sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69280#01981e72-0370-46ae-9528-4873c5245d43 | FLAKY         | 16/21    | upstream reliability is '99.38461538461539'. current run reliability is '76.19047619047619'. drift is 23.19414 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->
